qcodes/tests/dataset/test_measurement_context_manager.py

```diff
@@ -341,6 +341,17 @@ def action(lst, word):
 
 
 def test_subscriptions(experiment, DAC, DMM):
+    """
+    Test that subscribers are called at the moment that data is flushed to database
+
+    Note that for the purpose of this test, flush_data_to_database method is called explicitly instead of waiting for
+    the data to be flushed automatically after the write_period passes after a add_result call.
+
+    Args:
+        experiment (qcodes.dataset.experiment_container.Experiment) : qcodes experiment object
```

Comment:
For the main code base we try to move the type annotations out of the docstring and into the call signature, such that `mypy` can read them and tell us that we are being stupid. For tests I don't think it really matters much.